File: app/src/main/java/ch/bailu/aat/menus/MenuArray.java
package ch.bailu.aat.menus;

import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.graphics.drawable.Drawable;
import android.view.ActionProvider;
import android.view.ContextMenu;
import android.view.KeyEvent;
import android.view.Menu;
import android.view.MenuItem;
import android.view.SubMenu;
import android.view.View;

import java.util.ArrayList;


public class MenuArray implements Menu {
    private final ArrayList<Item> item = new ArrayList<>();
    private final Context context;

    public MenuArray(Context c) {
        context=c;
    }


    @Override
    public MenuItem add(CharSequence charSequence) {
        Item i = new Item(charSequence);
        item.add(i);
        return i;
    }

    @Override
    public MenuItem add(int i0, int id, int i2, CharSequence charSequence) {
        Item i = new Item(charSequence, id);
        item.add(i);
        return i;
    }

    @Override
    public int size() {
        return item.size();
    }


    @Override
    public MenuItem getItem(int i) {
        return item.get(i);
    }



    public CharSequence[] toStringArray() {
        CharSequence[] r = new CharSequence[item.size()];
        for (int i = 0; i < item.size(); i++) {
            r[i]= item.get(i).getTitle();
        }
        return r;
    }



    @Override
    public MenuItem add(int i) {
        return add(context.getString(i));
    }




    @Override
    public MenuItem add(int i, int id, int i2, int res) {
        return add(i, id, i2, context.getString(res));
    }

    @Override
    public SubMenu addSubMenu(CharSequence charSequence) {
        return null;
    }

    @Override
    public SubMenu addSubMenu(int i) {
        return null;
    }

    @Override
    public SubMenu addSubMenu(int i, int i1, int i2, CharSequence charSequence) {
        return null;
    }

    @Override
    public SubMenu addSubMenu(int i, int i1, int i2, int i3) {
        return null;
    }

    @Override
    public int addIntentOptions(int i, int i1, int i2, ComponentName componentName, Intent[] intents, Intent intent, int i3, MenuItem[] menuItems) {
        return 0;
    }

    @Override
    public void removeItem(int i) {

    }

    @Override
    public void removeGroup(int i) {

    }

    @Override
    public void clear() {

    }

    @Override
    public void setGroupCheckable(int i, boolean b, boolean b1) {

    }

    @Override
    public void setGroupVisible(int i, boolean b) {

    }

    @Override
    public void setGroupEnabled(int i, boolean b) {

    }

    @Override
    public boolean hasVisibleItems() {
        return false;
    }

    @Override
    public MenuItem findItem(int i) {
        return null;
    }



    @Override
    public void close() {

    }

    @Override
    public boolean performShortcut(int i, KeyEvent keyEvent, int i1) {
        return false;
    }

    @Override
    public boolean isShortcutKey(int i, KeyEvent keyEvent) {
        return false;
    }

    @Override
    public boolean performIdentifierAction(int i, int i1) {
        return false;
    }

    @Override
    public void setQwertyMode(boolean b) {

    }


    private class Item implements MenuItem {
        private CharSequence title;
        private final int id;


        public Item(CharSequence t) {
            this (t, 0);
        }

        public Item(CharSequence t, int i) {
            title = t;
            id = i;
        }


        @Override
        public MenuItem setTitle(CharSequence t) {
            title = t;
            return this;
        }


        @Override
        public MenuItem setTitle(int i) {
            return setTitle(context.getString(i));
        }

        @Override
        public CharSequence getTitle() {
            return title;
        }


        @Override
        public int getItemId() {
            return id;
        }

        @Override
        public int getGroupId() {
            return 0;
        }

        @Override
        public int getOrder() {
            return 0;
        }




        @Override
        public MenuItem setTitleCondensed(CharSequence charSequence) {
            return null;
        }

        @Override
        public CharSequence getTitleCondensed() {
            return null;
        }

        @Override
        public MenuItem setIcon(Drawable drawable) {
            return null;
        }

        @Override
        public MenuItem setIcon(int i) {
            return null;
        }

        @Override
        public Drawable getIcon() {
            return null;
        }

        @Override
        public MenuItem setIntent(Intent intent) {
            return null;
        }

        @Override
        public Intent getIntent() {
            return null;
        }

        @Override
        public MenuItem setShortcut(char c, char c1) {
            return null;
        }

        @Override
        public MenuItem setNumericShortcut(char c) {
            return null;
        }

        @Override
        public char getNumericShortcut() {
            return 0;
        }

        @Override
        public MenuItem setAlphabeticShortcut(char c) {
            return null;
        }

        @Override
        public char getAlphabeticShortcut() {
            return 0;
        }

        @Override
        public MenuItem setCheckable(boolean b) {
            return null;
        }

        @Override
        public boolean isCheckable() {
            return false;
        }

        @Override
        public MenuItem setChecked(boolean b) {
            return null;
        }

        @Override
        public boolean isChecked() {
            return false;
        }

        @Override
        public MenuItem setVisible(boolean b) {
            return null;
        }

        @Override
        public boolean isVisible() {
            return false;
        }

        @Override
        public MenuItem setEnabled(boolean b) {
            return null;
        }

        @Override
        public boolean isEnabled() {
            return false;
        }

        @Override
        public boolean hasSubMenu() {
            return false;
        }

        @Override
        public SubMenu getSubMenu() {
            return null;
        }

        @Override
        public MenuItem setOnMenuItemClickListener(OnMenuItemClickListener onMenuItemClickListener) {
            return null;
        }

        @Override
        public ContextMenu.ContextMenuInfo getMenuInfo() {
            return null;
        }

        @Override
        public void setShowAsAction(int i) {

        }

        @Override
        public MenuItem setShowAsActionFlags(int i) {
            return null;
        }

        @Override
        public MenuItem setActionView(View view) {
            return null;
        }

        @Override
        public MenuItem setActionView(int i) {
            return null;
        }

        @Override
        public View getActionView() {
            return null;
        }

        @Override
        public MenuItem setActionProvider(ActionProvider actionProvider) {
            return null;
        }

        @Override
        public ActionProvider getActionProvider() {
            return null;
        }

        @Override
        public boolean expandActionView() {
            return false;
        }

        @Override
        public boolean collapseActionView() {
            return false;
        }

        @Override
        public boolean isActionViewExpanded() {
            return false;
        }

        @Override
        public MenuItem setOnActionExpandListener(OnActionExpandListener onActionExpandListener) {
            return null;
        }
    }
}

File: aat-android/src/main/kotlin/ch/bailu/aat/menus/MenuArray.kt
package ch.bailu.aat.menus

import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.graphics.drawable.Drawable
import android.view.ActionProvider
import android.view.ContextMenu.ContextMenuInfo
import android.view.KeyEvent
import android.view.Menu
import android.view.MenuItem
import android.view.MenuItem.OnActionExpandListener
import android.view.SubMenu
import android.view.View

class MenuArray(private val context: Context) : Menu {
    private val item = ArrayList<Item>()
    override fun add(charSequence: CharSequence): MenuItem {
        val i = Item(context, charSequence)
        item.add(i)
        return i
    }

    override fun add(i0: Int, id: Int, i2: Int, charSequence: CharSequence): MenuItem {
        val i = Item(context, charSequence, id)
        item.add(i)
        return i
    }

    override fun size(): Int {
        return item.size
    }

    override fun getItem(i: Int): MenuItem {
        return item[i]
    }

    fun toStringArray(): Array<CharSequence?> {
        val r = arrayOfNulls<CharSequence>(item.size)
        for (i in item.indices) {
            r[i] = item[i].title
        }
        return r
    }

    override fun add(i: Int): MenuItem {
        return add(context.getString(i))
    }

    override fun add(i: Int, id: Int, i2: Int, res: Int): MenuItem {
        return add(i, id, i2, context.getString(res))
    }

    override fun addSubMenu(charSequence: CharSequence): SubMenu? {
        return null
    }

    override fun addSubMenu(i: Int): SubMenu? {
        return null
    }

    override fun addSubMenu(i: Int, i1: Int, i2: Int, charSequence: CharSequence): SubMenu? {
        return null
    }

    override fun addSubMenu(i: Int, i1: Int, i2: Int, i3: Int): SubMenu? {
        return null
    }

    override fun addIntentOptions(
        i: Int,
        i1: Int,
        i2: Int,
        componentName: ComponentName,
        intents: Array<Intent>,
        intent: Intent,
        i3: Int,
        menuItems: Array<MenuItem>
    ): Int {
        return 0
    }

    override fun removeItem(i: Int) {}
    override fun removeGroup(i: Int) {}
    override fun clear() {}
    override fun setGroupCheckable(i: Int, b: Boolean, b1: Boolean) {}
    override fun setGroupVisible(i: Int, b: Boolean) {}
    override fun setGroupEnabled(i: Int, b: Boolean) {}
    override fun hasVisibleItems(): Boolean {
        return false
    }

    override fun findItem(i: Int): MenuItem? {
        return null
    }

    override fun close() {}
    override fun performShortcut(i: Int, keyEvent: KeyEvent, i1: Int): Boolean {
        return false
    }

    override fun isShortcutKey(i: Int, keyEvent: KeyEvent): Boolean {
        return false
    }

    override fun performIdentifierAction(i: Int, i1: Int): Boolean {
        return false
    }

    override fun setQwertyMode(b: Boolean) {}
    private class Item @JvmOverloads constructor(
        private val context: Context,
        private var title: CharSequence?,
        private val id: Int = 0
    ) : MenuItem {
        override fun setTitle(t: CharSequence?): MenuItem {
            title = t
            return this
        }

        override fun setTitle(i: Int): MenuItem {
            return setTitle(context.getString(i))
        }

        override fun getTitle(): CharSequence? {
            return title
        }

        override fun getItemId(): Int {
            return id
        }

        override fun getGroupId(): Int {
            return 0
        }

        override fun getOrder(): Int {
            return 0
        }

        override fun setTitleCondensed(charSequence: CharSequence?): MenuItem {
            return this
        }

        override fun getTitleCondensed(): CharSequence? {
            return null
        }

        override fun setIcon(drawable: Drawable?): MenuItem {
            return this
        }

        override fun setIcon(i: Int): MenuItem {
            return this
        }

        override fun getIcon(): Drawable? {
            return null
        }

        override fun setIntent(intent: Intent?): MenuItem {
            return this
        }

        override fun getIntent(): Intent? {
            return null
        }

        override fun setShortcut(c: Char, c1: Char): MenuItem {
            return this
        }

        override fun setNumericShortcut(c: Char): MenuItem {
            return this
        }

        override fun getNumericShortcut(): Char {
            return Char(0)
        }

        override fun setAlphabeticShortcut(c: Char): MenuItem {
            return this
        }

        override fun getAlphabeticShortcut(): Char {
            return Char(0)
        }

        override fun setCheckable(b: Boolean): MenuItem {
            return this
        }

        override fun isCheckable(): Boolean {
            return false
        }

        override fun setChecked(b: Boolean): MenuItem {
            return this
        }

        override fun isChecked(): Boolean {
            return false
        }

        override fun setVisible(b: Boolean): MenuItem {
            return this
        }

        override fun isVisible(): Boolean {
            return false
        }

        override fun setEnabled(b: Boolean): MenuItem {
            return this
        }

        override fun isEnabled(): Boolean {
            return false
        }

        override fun hasSubMenu(): Boolean {
            return false
        }

        override fun getSubMenu(): SubMenu? {
            return null
        }

        override fun setOnMenuItemClickListener(onMenuItemClickListener: MenuItem.OnMenuItemClickListener?): MenuItem {
            return this
        }

        override fun getMenuInfo(): ContextMenuInfo? {
            return null
        }

        override fun setShowAsAction(i: Int) {}
        override fun setShowAsActionFlags(i: Int): MenuItem {
            return this
        }

        override fun setActionView(view: View?): MenuItem {
            return this
        }

        override fun setActionView(i: Int): MenuItem {
            return this
        }

        override fun getActionView(): View? {
            return null
        }

        override fun setActionProvider(actionProvider: ActionProvider?): MenuItem {
            return this
        }

        override fun getActionProvider(): ActionProvider? {
            return null
        }

        override fun expandActionView(): Boolean {
            return false
        }

        override fun collapseActionView(): Boolean {
            return false
        }

        override fun isActionViewExpanded(): Boolean {
            return false
        }

        override fun setOnActionExpandListener(onActionExpandListener: OnActionExpandListener?): MenuItem {
            return this
        }
    }
}
